source/js/main.js

```diff
@@ -379,6 +381,95 @@ let main = {
       );
     });
 
+    function profileCardSocialAnalytics(socialTwitter, socialLinkedIn) {
+      if (socialTwitter) {
+        socialTwitter.addEventListener(`click`, evt => {
+          evt.preventDefault();
+          ReactGA.event({
+            category: `profiles`,
+            action: `profile tap`,
+            label: `${document.title} ${name} twitter`,
+            transport: `beacon`
+          });
+        });
+      }
+
+      if (socialLinkedIn) {
+        socialLinkedIn.addEventListener(`click`, evt => {
+          evt.preventDefault();
+          ReactGA.event({
+            category: `profiles`,
+            action: `profile tap`,
+            label: `${document.title} ${name} linkedin`,
+            transport: `beacon`
+          });
+        });
+      }
+    }
+
+    function bindProfileCardAnalytics(profileCards) {
+      // event listener & GA
+      let bindAnalytics = (element, name) => {
+        element.addEventListener(`click`, evt => {
+          evt.preventDefault();
+          ReactGA.event({
+            category: `profiles`,
+            action: `profile tap`,
+            label: `${document.title} ${name} pulse profile`
+          });
+        });
+      };
+
+      // adding event listener for each headshot & nam
+      profileCards.forEach(card => {
+        let profileNameElement = card.querySelector(`.meta-block-name`);
+        let profileName = profileNameElement.textContent;
+        let profileHeadshotElement = card.querySelector(`.headshot-container`);
+
+        [(profileNameElement, profileHeadshotElement)].forEach(target =>
+          bindAnalytics(target, profileName)
+        );
+
+        let socialTwitter = card.querySelector(`.twitter`);
+        let socialLinkedIn = card.querySelector(`.linkedIn`);
+        profileCardSocialAnalytics(socialTwitter, socialLinkedIn);
+      });
+    }
+
+    // store profile cards
+    let profileCards = document.querySelectorAll(`.profiles .person-card`);
+
+    // checks for profile cards in the initial page load
+    if (profileCards.length > 0) {
+      bindProfileCardAnalytics(profileCards);
+    }
+
+    // And start listening for profile filter events,
+    // in case profile cards get updated.
+    document.addEventListener(`profiles:list-updated`, evt => {
+      evt.preventDefault();
+
+      //Profile Directory Filter Analytics
+      const filters = document.querySelectorAll(
```

Comment:
We don't need this to happen "whenever `profiles:list-updated` fires", we just need to run it once, as the filter buttons never get swapped out.

Reply:
✔️